embedded_fab/main.dart

```diff
@@ -0,0 +1,416 @@
+// Copyright 2018-present the Material Components for Flutter authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'package:flutter/material.dart';
+
+import 'bottom_items.dart';
+import 'colors.dart';
+import 'content.dart' as Content;
+import 'notch.dart';
+import 'painters.dart';
+import 'shape_type.dart';
+
+void main() => runApp(MyApp());
+
+class MyApp extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      debugShowCheckedModeBanner: false,
+      home: Home(),
+      theme: ThemeData.light().copyWith(scaffoldBackgroundColor: Colors.white),
+    );
+  }
+}
+
+class Home extends StatefulWidget {
+  Home({Key key}) : super(key: key);
+
+  @override
+  _HomeState createState() => _HomeState();
+}
+
+enum Layout {
+  AppBar,
+  Experimental,
+  SideFAB,
+}
+
+class _HomeState extends State<Home> {
+  Color _selectedColor = Colors.white;
+  int _selectedIndex = 0;
+  Layout _layout = Layout.Experimental;
+  ShapeType _selectedShape = ShapeType.Cut;
+
+  Color get _strokeColor => _selectedColor == blue500 ? Colors.white : null;
+
+  FloatingActionButton _traditionalFAB(onTap) {
+    return FloatingActionButton(
+      backgroundColor: Colors.white,
+      child: Image.asset('assets/icon.png'),
+      onPressed: onTap,
+    );
+  }
+
+  void _onItemTapped(int index) {
+    setState(() {
+      _selectedIndex = index;
+    });
+  }
+
+  void _setSelectedTile(value) {
```

Comment:
This param should probably be typed.

```suggestion
  void _setSelectedTile(ShapeType value) {
```

Reply:
Same for the method below.

Reply:
Done